src/python/pants/backend/python/tasks/python_run.py

```diff
@@ -51,5 +52,6 @@ def execute(self):
             msg = '{cmdline} ... exited non-zero ({code})'.format(cmdline=cmdline, code=result)
             raise TaskError(msg, exit_code=result)
         except KeyboardInterrupt:
-          po.send_signal(signal.SIGINT)
+          # The process may still have exited, even if we were interrupted.
+          safe_kill(po.pid, signal.SIGINT)
```

Comment:
You can leave this in this PR, but a reminder: please do not conflate independent changes in one PR like this in the future. If this line causes a bug, or if the rest of the PR contains a bug, they won't be able to be cherry-picked independently. And it also makes reviewing more challenging.

Reply:
This was necessary to pass testing. It may not be necessary now with the removal of the signal handlers.

Reply:
Again, I'm fine with leaving it... just a note for future PRs.

Reply:
Reverted!